src/baldr/graphreader.cc

```diff
@@ -732,6 +733,16 @@ std::pair<GraphId, GraphId> GraphReader::GetDirectedEdgeNodes(const GraphTile* t
   return std::make_pair(start_node, end_node);
 }
 
+std::string GraphReader::edge_shape(const valhalla::baldr::GraphId& edgeid) {
+  const baldr::GraphTile* t_debug = GetGraphTile(edgeid);
```

Comment:
Perhaps a null-pointer check here and for `directedEdge` would be good.

Reply:
I though about it too. There are two possible ways of doing it. 
1. as you pointed out, check nullptr of the t_debug. 
2. is to check the edge_id valid.

It may arise some question that, does any valid edge_id has an associate tile*? if not, what exceptions do we throw? May need to dig out that

Reply:
All the exceptions I've seen are simply `std::runtime_error("Helpful text")`. That should be good enough since this would only be for testing purposes.

Reply:
After reading the rest of the code in the .cc file, I found out, the rest code was assuming if edgeid is valid, there should be a tile* associate with it. No check nullptr was done. Thus, I don't check it here too for consistancy.

Reply:
no you have to do the `nullptr` check. look at the other convenience methods in this class such as: https://github.com/valhalla/valhalla/blob/master/src/baldr/graphreader.cc#L441

anywhere where you get a tile, you may get back a nullpointer. there are some cases where you know for sure that you cannot get back a nullpointer and in those cases i could see not checking, but in this case, you must do so.

Reply:
You can still get a segfault if someone called this with a bad/invalid `edgeid`. So maybe add a validity check for `edged`? I'll leave the final decision to you and others more informed can chime in.